### PR TITLE
Makefile fixes / JUTNameTab

### PR DESCRIPTION
**Pull request by magcius**

### CODE REVIEW

File: libs/JSystem/J3DGraphBase/J3DPacket.cpp [hunk 1]
```diff
@@ -160,68 +107,55 @@ extern "C" extern u8 __GDCurrentDL[4];
...
-/* 803125E4-80312604 30CF24 0020+00 0/0 1/1 0/0 .text
- * setSingleDisplayList__17J3DDisplayListObjFPvUl               */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::setSingleDisplayList(void* param_0, u32 param_1) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/setSingleDisplayList__17J3DDisplayListObjFPvUl.s"
+void J3DDisplayListObj::setSingleDisplayList(void* pDLData, u32 size) {
+    mCapacity = (size + 0x1F) & ~0x1F;
```
Comment (lepelog) [inline libs/JSystem/J3DGraphBase/J3DPacket.cpp:144]:
```suggestion
    mCapacity = ALIGN_NEXT(size, 0x20);
```
For that macro you need to import "global.h"

File: libs/JSystem/J3DGraphBase/J3DPacket.cpp [hunk 2]
```diff
@@ -160,68 +107,55 @@ extern "C" extern u8 __GDCurrentDL[4];
...
-/* 80312508-8031256C 30CE48 0064+00 1/1 1/1 0/0 .text newSingleDisplayList__17J3DDisplayListObjFUl
- */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::newSingleDisplayList(u32 param_0) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/newSingleDisplayList__17J3DDisplayListObjFUl.s"
+J3DError J3DDisplayListObj::newSingleDisplayList(u32 capacity) {
+    mCapacity = (capacity + 0x1F) & ~0x1F; // align
```
Comment (lepelog) [inline libs/JSystem/J3DGraphBase/J3DPacket.cpp:121]: ALIGN_NEXT macro

File: libs/JSystem/J3DGraphBase/J3DPacket.cpp [hunk 3]
```diff
@@ -160,68 +107,55 @@ extern "C" extern u8 __GDCurrentDL[4];
...
 
-/* 80312488-80312508 30CDC8 0080+00 1/1 1/1 0/0 .text newDisplayList__17J3DDisplayListObjFUl */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-asm void J3DDisplayListObj::newDisplayList(u32 param_0) {
-    nofralloc
-#include "asm/JSystem/J3DGraphBase/J3DPacket/newDisplayList__17J3DDisplayListObjFUl.s"
+J3DError J3DDisplayListObj::newDisplayList(u32 capacity) {
+    mCapacity = (capacity + 0x1F) & ~0x1F; // align
```
Comment (lepelog) [inline libs/JSystem/J3DGraphBase/J3DPacket.cpp:111]: ALIGN_NEXT macro

File: include/JSystem/JUtility/JUTNameTab.h [hunk 4]
```diff
@@ -3,4 +3,34 @@
...
+struct ResNTAB {
+    u16 mEntryNum;
+    u16 mPad0;
+    struct Entry {
+        u16 mKeyCode;
+        u16 mOffs;
+    } mEntries[1];
+
+    inline const char* getName(u32 index) const {
+        return ((const char*)mEntries) + ((u16*)mEntries)[index * 2 + 1] - 4;
```
Comment (lepelog) [inline include/JSystem/JUtility/JUTNameTab.h:15]:
```suggestion
        return ((const char*)mEntries) + mEntries[index].mOffs - 4;
```
Or is there a reason to write it the other way?

Comment (magcius) [reply include/JSystem/JUtility/JUTNameTab.h:15]: This was probably nonsense I was writing when trying to get a match for the loop. This branch is a couple months old.

File: libs/JSystem/JUtility/JUTNameTab.cpp [hunk 5]
```diff
@@ -39,69 +20,62 @@ extern "C" void calcKeyCode__10JUTNameTabCFPCc();
...
-asm void JUTNameTab::setResource(ResNTAB const* param_0) {
-    nofralloc
-#include "asm/JSystem/JUtility/JUTNameTab/setResource__10JUTNameTabFPC7ResNTAB.s"
+#if NON_MATCHING
+s32 JUTNameTab::getIndex(const char* pName) const {
+    const ResNTAB::Entry *pEntry = mpNameTable->mEntries;
+    u16 keyCode = calcKeyCode(pName);
+
+    // Getting mwcc to emit an rlwinm. instead of a loop counter appears to be challenging
+    for (s32 i = 0; i < mNameNum; i++, pEntry++)
```
[the remark this answers is not in the document]
Comment (magcius) [reply libs/JSystem/JUtility/JUTNameTab.cpp:50]: Nice, this worked.

**Review by lepelog — APPROVED** [review]

**Pull request merged**